File: 1458-sort-integers-by-the-number-of-1-bits/sort-integers-by-the-number-of-1-bits.cpp
class Solution {
public:
    vector<int> sortByBits(vector<int>& arr) {
        priority_queue<pair<int,int>,vector<pair<int,int>>,greater<pair<int,int>>> pq;
        for(int i=0;i<arr.size();i++){
            pq.push({__builtin_popcount(arr[i]),arr[i]});
        }
        vector<int> ans;
        while(!pq.empty()){
            int val=pq.top().second;
            pq.pop();
            ans.push_back(val);

        }
        return ans;
    }
};

File: 1458-sort-integers-by-the-number-of-1-bits/sort-integers-by-the-number-of-1-bits.java
import java.util.*;

class Solution {
    public int[] sortByBits(int[] arr) {
        PriorityQueue<int[]> pq = new PriorityQueue<>((a, b) -> {
            if (a[0] == b[0]) return a[1]-b[1];
            return a[0]-b[0];
        });

        for (int i = 0; i < arr.length; i++) {
            pq.add(new int[]{Integer.bitCount(arr[i]), arr[i]});
        }

        int[] ans = new int[arr.length];
        int index = 0;
        while (!pq.isEmpty()) {
            int val = pq.peek()[1];
            pq.poll();
            ans[index++] = val;
        }

        return ans;
    }
}